src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -145,7 +146,7 @@ public SQLiteStore(String directory, Manager manager, StoreDelegate delegate) {
         }
         this.path = new File(directory, kDBFilename).getPath();
         this.manager = manager;
-        this.storageEngine = null;
+        this.storageEngine = createStorageEngine();
```

Comment:
I prefer your second option. Because storageEgnine could be null when `derivePBKDF2SHA256Key()` is called.